File: go/leetcode/83. Remove Duplicates from Sorted List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
    if head == nil || head.Next == nil{return head}
    
    slow := head;
    fast := slow.Next;
    slow.Next = nil;
    
    for fast != nil {
        if fast.Val != slow.Val {
            slow.Next = fast;
            slow = fast;
            fast = fast.Next;
            slow.Next = nil;
            continue;
        }
        fast = fast.Next;
    }
    
    return head;
}

File: javascript/leetcode/83. Remove Duplicates from Sorted List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
  if (!head || !head.next) return head;

  let slow = head;
  let fast = slow.next;
  slow.next = null;

  while (fast) {
    if (fast.val !== slow.val) {
      slow.next = fast;
      slow = fast;
      fast = fast.next;
      slow.next = null;
      continue;
    }
    fast = fast.next;
  }

  return head;
};
